Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,305 @@
+                    Converters = new List<JsonConverter> { new Newtonsoft.Json.Converters.StringEnumConverter() } // Use string enum values instead of numerical.
+                });
+
+            return JObject.Parse(stringSchema);
+        }
+
+        private static List<JObject> GetSchemaFromPath(string path, out List<string> errorList, bool validateAsDictionary)
+        {
+            errorList = new List<string>();
+
+            try
+            {
+                string schema = File.ReadAllText(path);
+
+                List<JObject> res = ConvertStringToJObjectList(schema, out List<string> convertError, validateAsDictionary);
+                errorList = convertError;
+
+                return res;
+            }
+            catch (Exception e)
+            {
+                errorList.Add(e.Message);
+
+                return new List<JObject>();
+            }
+        }
+
+        private static List<JObject> GetAllSchemasInDirectory(string path, out List<string> errorList, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+            errorList = new List<string>();
+
+            try
+            {
+                string[] Files = Directory.GetFiles(path);
+                List<object> schemalist = new List<object>();
+
+                if (validateAsDictionary) {
+                    Dictionary<string, ForgeTree> combinedDictionary = new Dictionary<string, ForgeTree>();
+                    
+                    foreach (string file in Files) 
+                    {
+                        string schema = File.ReadAllText(file);
+                        Dictionary<string, ForgeTree> schemaDictionary = JsonConvert.DeserializeObject<Dictionary<string, ForgeTree>>(schema);
+                        
+                        foreach (KeyValuePair<string, ForgeTree> item in schemaDictionary) 
+                        {
+                            combinedDictionary.Add(item.Key, item.Value);
+                        }
+                    }
+
+                    return new List<JObject> { SerializeToJObject(combinedDictionary) };
+                }
+                else
+                {
+                    foreach (string file in Files)
+                    {
+                        List<JObject> schemasInFile = GetSchemaFromPath(file, out errorList, validateAsDictionary);
+
+                        if (errorList.Count > 0)
+                        {
+                            break;
+                        }
+
+                        schemasInFile.ForEach(n => schemaList.Add(n));
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                errorList.Add(e.Message);
+            }
+
+            return schemaList;
+        }
+
+        private static bool CheckConvertErrorAndValidate(Object rules, ref List<string> errorList, List<JObject> schemaList)
```

Comment:
>Object [](http://example.com/codeflow?start=57&length=6)

It is recommended to always use the language-specific (C#) types instead of the .Net types. Basically this is the lowercase types, such as object, string, int. Do not use Object, String, Int32, etc..